src/core/public/http/types.ts

```diff
@@ -276,12 +302,13 @@ export interface IHttpFetchError extends Error {
 }
 
 /** @public */
-export interface HttpErrorResponse extends IHttpResponse {
+export interface HttpErrorResponse extends HttpResponse {
+  request: Readonly<Request>;
   error: Error | IHttpFetchError;
 }
 /** @public */
 export interface HttpErrorRequest {
```

Comment:
outside of the scope of the PR probably, but isn't `HttpRequestError` more appropriate?

Reply:
This type is only used when a request interceptor throws an error while processing a request. Maybe `HttpInterceptorRequestError` would make more sense?

Reply:
it's very minor, but your naming makes more sense than the actual one yes